### PR TITLE
Enhancement: add `emit_empty_json_arrays` config so `[]` can be encoded for empty results

### PR DESCRIPTION
**Pull request by powersjcb**
It looks like the default behavior of `json.Marshal` leaks go language implementation details. Empty slices which are initialized with the syntax, `var items []Data` are backed by `nil`. The golang stdlib marshaler encodes `nil` slices as `null`.

Empty slices which are initialized with the syntax `items := make([]Data, 0)` are backed by a pointer to an empty slice. The marshaller encodes empty slices as `[]`.

There's unfortunately no way to add constructor checks or json tag annotations with the current go runtime. (It appears the only other way to resolve this issue would be to use a 3rd party marshaller.)

This change adds a configuration that allows for control of the way slices are constructed and causes `json.Marshal` to encode `[]` for empty results.

Resolves Issue #526

### CODE REVIEW

**Comment by kyleconroy** [comment]
Thanks for fixing the bug your found, really appreciate it.

I'm still mulling over if I want to make this change. I understand the bug, but want to make sure it doesn't break existing code for others. It's always scary when sqlc generate creates a diff.

**Comment by powersjcb** [comment]
Yeah, this change will break a client implementations in both of these cases:
- client explicitly handles the `null` case with something other than skipping iteration over a for loop
- client assumes that `items.length > 0` if `items !== null`

I feel like the second assumption is unlikely, but the first case is plausible.


Have any thoughts about making this a config change or a sem version bump?

File: examples/booktest/postgresql/jsonemptyarrays/db_test.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// +build examples
```
Comment (powersjcb) [inline examples/booktest/postgresql/jsonemptyarrays/db_test.go:1]: Any opinions about this being nested under `./postgresql/json` vs being it's own top level test dir?

**Comment by powersjcb** [comment]
Updated the PR to make this an optional config.

**Comment by kyleconroy** [comment]
@powersjcb ahh, I’m so sorry this didn’t get merged. I got this PR confused with a very similar one (https://github.com/kyleconroy/sqlc/pull/551) which I ended up merging.

If you’d like, you can update this PR to use that configuration option, but for now I think the best action is to close this PR.

Sorry again for the mistake, I don’t want you to feel like your contributions aren’t welcome.

**Pull request closed without merging**